client/src/app/store/chat/chat.model.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    ChatPreview_,
+    StoredMessage_,
+    MessagePreview_,
+    ReceivedFriendshipInvitation,
+    ChatGroup,
+    ChatGroupDetails,
+    Friendship,
+    FriendshipDetails,
+    UserDetails,
+    SentFriendshipInvitation,
+} from 'src/shared/index.model';
+import { HttpSuccessResponse } from '../app.model';
+
+export type MessagePreview = MessagePreview_<'client'>;
+export type StoredMessage = StoredMessage_<'client'>;
+export type ChatPreview = ChatPreview_<'client'>;
+
+export class ChatsState {
+    activeChatId: string | null = null;
+
+    chatPreviews: ChatPreview[] = [];
+    loadingChatPreviews: boolean = false;
+
+    // @TODO: implement details view
+    // chatGroups: Record<string, ChatGroup> = {};
+    // chatGroupsDetails: Record<string, ChatGroupDetails> = {};
+    // friendships: Record<string, Friendship> = {};
+    // friendshipsDetails: Record<string, FriendshipDetails> = {};
+    // loadingDetails: boolean = false;
+
+    messagesByChat: Record<string, StoredMessage[]> = {};
+    loadingActiveChatMessages = false;
+
+    users: Record<string, UserDetails>;
+
+    receivedInvitations: {
+        pending: ReceivedFriendshipInvitation[];
+        accepted: ReceivedFriendshipInvitation[];
+        declined: ReceivedFriendshipInvitation[];
+        loading: boolean;
+    } = {
+        // @TODO: maybe let them default to null
+        pending: [],
+        accepted: [],
+        declined: [],
```

Comment:
Lets default these to `null` and not show a counter badge when they are stillt `null`